tests/test_samples.py

```diff
@@ -0,0 +1,283 @@
+        self._temp_file = NamedTemporaryFile(
+            mode="w+",
+            dir=os.path.dirname(sample_filename),
+            delete=False)
+        self._temp_file.close()
+        os.chmod(self._temp_file.name, 0o777)
+        self.base_filename = sample_filename
+        self.write_file_line(full_copy=True)
+
+    def write_file_line(self, target=None, replacement=None, full_copy=False):
+        if full_copy:
+            base_filename = self.base_filename
+            target_filename = self._temp_file.name
+        else:
+            base_filename = self._temp_file.name
+            target_filename = base_filename + "new"
+
+        with open(base_filename, 'r') as base_file:
+            with open(target_filename, 'w+') as new_sample_file:
+                for line in base_file:
+                    if target != None and replacement != None:
+                        if line.startswith(target):
+                            line = replacement
+                    new_sample_file.write(line)
+
+        if not full_copy:
+            os.remove(base_filename)
+            os.rename(target_filename, base_filename)
+            os.chmod(base_filename, 0o777)
+
+
+    def __del__(self):
+        self.temp_file.close()
+        os.remove(self._temp_file.name)
+
+
+class TestSamples(BaseClientTest):
+
+    SAMPLE_FOLDER = str(os.path.dirname(
+        os.path.dirname(
+            os.path.abspath(__file__)
+        )
+    ).replace("\\", "/")) + "/sample"
+
+    BASIC_FOLDER = SAMPLE_FOLDER + "/basic"
+    ADVANCED_FOLDER = SAMPLE_FOLDER + "/advanced"
+
+
+    def test_basicfirstref_example(self):
+        # Modify sample file to include necessary sample data
+        sample_filename = self.BASIC_FOLDER + "/basic_first_ref_example.py"
+        temp_sample_file = TempSampleFile(sample_filename)
+
+        target_line = "FILE_MD5 = "
+        replacement_line = target_line + "\"" + FILE_NOTEPAD_EXE_HASH_DICT[HashType.MD5] + "\"\n"
+        temp_sample_file.write_file_line(
+            target_line,
+            replacement_line
+        )
+
+        target_line = "FILE_SHA1 = "
+        replacement_line = target_line + "\"" + FILE_NOTEPAD_EXE_HASH_DICT[HashType.SHA1] + "\"\n"
+        temp_sample_file.write_file_line(
+            target_line,
+            replacement_line
+        )
+
+        target_line = "FILE_SHA256 = "
+        replacement_line = target_line + "\"" + FILE_NOTEPAD_EXE_HASH_DICT[HashType.SHA256] + "\"\n"
+        temp_sample_file.write_file_line(
+            target_line,
+            replacement_line
+        )
+
+        with self.create_client(max_retries=0) as dxl_client:
+            # Set up client, and register mock service
+            dxl_client.connect()
+
+            with MockTieServer(dxl_client):
+                mock_print = run_sample(temp_sample_file.temp_file.name)
+
+                mock_print.assert_any_call(
+                    StringContains("Systems that have referenced the file:")
```

Comment:
Might be good to also have it assert that some info from the response appeared in the output - like some of the system GUIDs?

Reply:
Done, now verifies all expected GUIDs present.